core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreReaderThread.java

```diff
@@ -79,57 +71,49 @@ public KafkaStoreReaderThread(ZkUtils zkUtils,
     this.storeUpdateHandler = storeUpdateHandler;
     this.serializer = serializer;
     this.localStore = localStore;
-    this.commitInterval = commitInterval;
     this.noopKey = noopKey;
 
-    offsetInSchemasTopic = offsetOfLastConsumedMessage(zkUtils, groupId, topic);
-    log.info("Initialized the consumer offset to " + offsetInSchemasTopic);
     Properties consumerProps = new Properties();
-    consumerProps.put("group.id", this.groupId);
-    consumerProps.put("client.id", "KafkaStore-reader-" + this.topic);
-    consumerProps.put("zookeeper.connect", kafkaClusterZkUrl);
-    consumerProps.put("auto.offset.reset", "smallest");
-    consumerProps.put("auto.commit.enable", "false");
-    consumer = new ZookeeperConsumerConnector(new ConsumerConfig(consumerProps));
-    Map<String, Integer> kafkaStreamConfig = new HashMap<String, Integer>();
-    kafkaStreamConfig.put(topic, 1);
-    Map<String, List<KafkaStream<byte[], byte[]>>> streams =
-        consumer.createMessageStreams(kafkaStreamConfig);
-    List<KafkaStream<byte[], byte[]>> streamsForTheLogTopic = streams.get(topic);
-    // there should be only one kafka partition and hence only one stream
-    if (streamsForTheLogTopic != null && streamsForTheLogTopic.size() != 1) {
+    consumerProps.put(ConsumerConfig.GROUP_ID_CONFIG, this.groupId);
+    consumerProps.put(ConsumerConfig.CLIENT_ID_CONFIG, "KafkaStore-reader-" + this.topic);
+
+    consumerProps.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapBrokers);
+    consumerProps.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+    consumerProps.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+    consumerProps.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
+            org.apache.kafka.common.serialization.ByteArrayDeserializer.class);
+    consumerProps.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+            org.apache.kafka.common.serialization.ByteArrayDeserializer.class);
+    this.consumer = new KafkaConsumer<>(consumerProps);
+
+    int partitionCount = this.consumer.partitionsFor(this.topic).size();
```

Comment:
It looks like partitionsFor() returns null instead of an empty collection when the topic is unknown (I should probably clarify that in the KafkaConsumer docs). It might be safest to check both for null and size() < 1.